tests/activity_api_update_test.go

```diff
@@ -0,0 +1,340 @@
+	"go.temporal.io/server/common/testing/testvars"
+	"go.temporal.io/server/tests/testcore"
+	"google.golang.org/protobuf/types/known/durationpb"
+	"google.golang.org/protobuf/types/known/fieldmaskpb"
+)
+
+type ActivityApiUpdateClientTestSuite struct {
+	testcore.ClientFunctionalSuite
+	tv                     *testvars.TestVars
+	initialRetryInterval   time.Duration
+	scheduleToCloseTimeout time.Duration
+	startToCloseTimeout    time.Duration
+
+	activityRetryPolicy *temporal.RetryPolicy
+}
+
+func (s *ActivityApiUpdateClientTestSuite) SetupSuite() {
+	s.ClientFunctionalSuite.SetupSuite()
+	s.OverrideDynamicConfig(dynamicconfig.ActivityAPIsEnabled, true)
+	s.tv = testvars.New(s.T()).WithTaskQueue(s.TaskQueue()).WithNamespaceName(namespace.Name(s.Namespace()))
+}
+
+func (s *ActivityApiUpdateClientTestSuite) SetupTest() {
+	s.ClientFunctionalSuite.SetupTest()
+
+	s.initialRetryInterval = 10 * time.Second
+	s.scheduleToCloseTimeout = 30 * time.Second
+	s.startToCloseTimeout = 10 * time.Second
+
+	s.activityRetryPolicy = &temporal.RetryPolicy{
+		InitialInterval:    s.initialRetryInterval,
+		BackoffCoefficient: 1,
+		MaximumInterval:    10 * time.Second,
+	}
+}
+
+func TestActivityApiUpdateClientTestSuite(t *testing.T) {
+	s := new(ActivityApiUpdateClientTestSuite)
+	suite.Run(t, s)
+}
+
+func (s *ActivityApiUpdateClientTestSuite) waitForChan(ctx context.Context, ch chan struct{}) {
+	s.T().Helper()
+	select {
+	case <-ch:
+	case <-ctx.Done():
+		s.FailNow("context timeout")
+	}
+}
+
+type (
+	ActivityFunctions func() (string, error)
+	WorkflowFunction  func(context2 workflow.Context) (string, error)
+)
+
+func (s *ActivityApiUpdateClientTestSuite) makeWorkflowFunc(activityFunction ActivityFunctions) WorkflowFunction {
+	return func(ctx workflow.Context) (string, error) {
+
+		var ret string
+		err := workflow.ExecuteActivity(workflow.WithActivityOptions(ctx, workflow.ActivityOptions{
+			ActivityID:             "activity-id",
+			DisableEagerExecution:  true,
+			StartToCloseTimeout:    s.startToCloseTimeout,
+			ScheduleToCloseTimeout: s.scheduleToCloseTimeout,
+			RetryPolicy:            s.activityRetryPolicy,
+		}), activityFunction).Get(ctx, &ret)
+		return "done!", err
+	}
+}
+
+func (s *ActivityApiUpdateClientTestSuite) TestActivityUpdateApi_ChangeRetryInterval() {
+	ctx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer cancel()
+
+	activityUpdated := make(chan struct{})
+
+	var activityCompleted atomic.Int32
+	activityFunction := func() (string, error) {
+		if activityCompleted.Load() == 0 {
+			activityErr := errors.New("bad-luck-please-retry")
+			activityCompleted.Add(1)
+			return "", activityErr
+		}
+
+		s.waitForChan(ctx, activityUpdated)
+		activityCompleted.Add(1)
+		return "done!", nil
+	}
+
+	workflowFn := s.makeWorkflowFunc(activityFunction)
+
+	s.Worker().RegisterWorkflow(workflowFn)
+	s.Worker().RegisterActivity(activityFunction)
+
+	wfId := "functional-test-activity-update-api"
```

Comment:
I'd introduce some randomness here just in case you want to be able to rerun this test without restarting the server.

```suggestion
	wfId := testcore.RandomizeStr("wfid-"+s.T().Name())
```

Reply:
added